eventuate-log-cassandra/src/main/scala/com/rbmhtechnology/eventuate/log/cassandra/CassandraEventLog.scala

```diff
@@ -136,7 +137,7 @@ class CassandraEventLog(id: String, aggregateIndexing: Boolean) extends EventLog
 
   override def writeEventLogClockSnapshot(clock: EventLogClock): Future[Unit] = {
     import context.dispatcher
-    indexStore.writeEventLogClockSnapshotAsync(clock).map(_ => ())
+    updateIndex(clock).flatMap(_ => indexStore.writeEventLogClockSnapshotAsync(clock)).map(_ => ())
```

Comment:
What do you mean it won't happen? As you pointed out it will happen, if `aggregateIndexing` is true and disaster recovery is invoked, right? But I would tend to agree, this should not affect the performance considerably, so I would tend to keep it as it is to avoid additional complexity.